core/src/main/java/org/kiwix/kiwixmobile/core/base/BasePresenter.kt

```diff
@@ -15,30 +15,26 @@
  * along with this program. If not, see <http://www.gnu.org/licenses/>.
  *
  */
-package org.kiwix.kiwixmobile.core.base;
+package org.kiwix.kiwixmobile.core.base
 
-import io.reactivex.disposables.CompositeDisposable;
+import io.reactivex.disposables.CompositeDisposable
+import org.kiwix.kiwixmobile.core.base.BaseContract.Presenter
+import org.kiwix.kiwixmobile.core.base.BaseContract.View
 
 /**
  * All presenters should inherit from this presenter.
  */
-
-public abstract class BasePresenter<T extends BaseContract.View>
-  implements BaseContract.Presenter<T> {
-
-  protected final CompositeDisposable compositeDisposable = new CompositeDisposable();
-  protected T view;
-
-  @Override
-  public void attachView(T view) {
-    this.view = view;
+abstract class BasePresenter<T : View<*>?> : Presenter<T> {
+  val compositeDisposable = CompositeDisposable()
+  var view: T? = null
+  override fun attachView(view: T) {
```

Comment:
A new line between the variable declaration and the beginning of the function would be good.

Reply:
Yes, it is good for code readability, but as the code increases in size, it is not recommended as it will increase the lines of code for bigger projects and hence can be a problem in future. By avoiding this spaces, we can decrease the lines of codes drastically in a huge code.

Reply:
@yash982000 I don't think that having readable code is bad. Removing extra lines between different sections of code and different function declarations will only make the code look more congested and untidy.

Reply:
@yash982000 there should be a space between properties and functions, 1 space isn't going to make this file unreadable even if it does bloat massively.

There should also be spaces between functions which is maybe what ye are talking about? Regardless put spaces between functions. Way more readable

> Code is written to be read by humans and incidentally machines

Reply:
@akkywadhwa add a space between the properties and the function

Reply:
Yes, sir. Actually, I made a mistake by misreading it. There should be always a space between properties and functions. I was thinking about some other space.